src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -13,472 +13,528 @@ namespace nbl::asset
+            getParentTypeStore()->memberNames()(m_memPool.data())[entry.memberIndex] = addString(comp.get_member_name(entry.parentTypeID,entry.memberIndex));
+            const auto& parentType = comp.get_type(entry.parentTypeID);
+            getParentTypeStore()->memberSizes()(m_memPool.data())[entry.memberIndex] = comp.get_declared_struct_member_size(parentType,entry.memberIndex);
+            getParentTypeStore()->memberOffsets()(m_memPool.data())[entry.memberIndex] = comp.type_struct_member_offset(parentType,entry.memberIndex);
+            getParentTypeStore()->memberStrides()(m_memPool.data())[entry.memberIndex] = getTypeStore()->isArray() ? comp.type_struct_member_array_stride(parentType,entry.memberIndex):0u;
+//            comp.get_declared_struct_size();
+//            comp.get_declared_struct_size_runtime_array();
+#if 0 
+    for (uint32_t m = 0u; m < memberCnt; ++m)
+    {
+        MembT& member = dstMembers.array[m];
+        const spirv_cross::SPIRType& mtype = comp.get_type(allMembersTypes[m]);
 
-                //all formats returned by glslType2E_FORMAT() already are multiple-of-4 bytes so no need to pad
-                reloffset += getTexelOrBlockBytesize(static_cast<E_FORMAT>(attr.format));
+        member.rowMajor = _comp.get_member_decoration(_parentType.self, m, spv::DecorationRowMajor); // ?
 
-                vtxInput.enabledAttribFlags |= (1u << io.location);
-            }
+            //if (member.mtxColCnt > 1u)
+            //    member.mtxStride = _comp.type_struct_member_matrix_stride(_parentType, m);
+#endif
         }
-        vtxInput.enabledBindingFlags = vtxInput.enabledAttribFlags;
 
-        for (uint32_t i = 0u; i < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; ++i)
+        // found in cache, then don't need to fill out the rest
+        if (found!=typeCache.end())
+            continue;
+
+        getTypeStore()->count = addCounts(type.array.size(),type.array.data(),type.array_size_literal.data());
+        if (!entry.isRoot())
         {
-            if (vtxInput.enabledBindingFlags & (1u << i))
-                vtxInput.bindings[i].stride = reloffset;
-        }
-    }
+            const auto& parentType = comp.get_type(entry.parentTypeID);
+            auto memberId = parentType.member_types[entry.memberIndex];
+            auto memberBaseType = comp.get_type(memberId).basetype;
 
-    //all except vertex input are defaulted
-    SBlendParams blending;
-    SPrimitiveAssemblyParams primAssembly;
-    SRasterizationParams raster;
+            auto typeName = getBaseTypeNameFromEnum(spvcrossType2E_TYPE(memberBaseType));
 
-    auto layout = createApproximatePipelineLayoutFromIntrospection_impl(introspections, { infos.begin(),infos.end() });
+            getTypeStore()->typeName = addString(getBaseTypeNameFromEnum(spvcrossType2E_TYPE(memberBaseType)));
+        }
+        else
+        {
+            getTypeStore()->typeName = addString("TODO");
```

Comment:
You could actually find the name of a type because whenever one defines a type in SPIR-V they use `OpType...` which gives them a result ID (integer)
```glsl

in vec4 color1;
in vec4 multiplier;
noperspective in vec4 color2;
out vec4 color;

struct S {
    bool b;
    vec4 v[5];
    int i;
};

uniform blockName {
    S s;
    bool cond;
};
```
this is the SPIR-V that generates them
```
          %6 = OpTypeFloat 32                         ; 32-bit float
          %7 = OpTypeVector %6 4                      ; vec4

         %10 = OpConstant %6 1
         %11 = OpConstant %6 2

         %13 = OpTypeInt 32 0                         ; 32-bit int, sign-less
         %14 = OpConstant %13 5
         %15 = OpTypeArray %7 %14
         %16 = OpTypeInt 32 1
         %17 = OpTypeStruct %13 %15 %16
         %18 = OpTypeStruct %17 %13
```
and in the debug info you'll have
```
               OpName %17 "S"
               OpMemberName %17 0 "b"
               OpMemberName %17 1 "v"
               OpMemberName %17 2 "i"
               OpName %18 "blockName"
               OpMemberName %18 0 "s"
               OpMemberName %18 1 "cond"
```

Note that depending on the compiler, anything thats an `OpTypeFloat`, `OpTypeInt`, `OpTypeVector` or `OpTypeArray` might not get debug names so we should probably build outselves.

You don't need to do all this right now, because I want you to progress to other tasks, but would be good to write it up here and open a GH issue about it for later fixing.

Reply:
add this as a block comment  on this line

Reply:
please add the concents of the first comment as block comment on this line